#  LSM 3243

> By the end of the course, you should be able to
> 1. Describe protein backbone and side-chain conformations in terms of dihedral angles and interactions
> 2. Understand the factors that determine protein conformations
> 3. Know common nucleic acid conformations and to understand
>    the interactions that stabilize such ordered conformations
> 4. Know the composition of cell membrane and understand unique features of membrane proteins
> 5. Understand the structures of micelles and bilayers
> 6. Understand how small molecules are transported across membranes
> 7. Understand basic concepts of spectroscopic techniques: circular dichroism, fluorescence and nuclear magnetic resonance
> 8. Be knowledgeable to the application of these techniques to life sciences
> 9. State transition
> 10. Protein folding
> 11. Protein-ligand interaction



## Week1 Protein Structure

### Introduction

1. Only L-amino acids are found in natural proteins.

   Look along the $\ce{H-C_\alpha}$ bond, seeing <u>CO-R-N in clockwise</u> indicates L.

2. Nomenclature of main/side chain atoms: $\alpha, \beta, \gamma\dots$

### Polypeptide chain

#### peptide bonds

- N is partially positive (resonance?) while O is partially negative.
- The 6 atoms are coplanar!

#### torsion angle

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/1_1.png" alt="1_1" style="zoom: 20%;" />

- Look through B-C, if B-A (the nearer) needs to rotate <u>clockwise</u> to reach superposition with C-D, the angle is positive.
- But in which direction we look through doesn't matter because the result is <u>the same</u>.


#### backbone dihedral angles

> - $\phi : \ce{N-C_{\alpha}}$
> - $\psi : \ce{C_O-C_{\alpha}}$
> - $\omega : \ce{C_O-N}$

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/1_2.png" alt="1_1" style="zoom: 12%;" />

The protein structure is almost determined by these angles.

##### $\omega$ angle

It is usually fixed.

- For proline, groups in *trans* is less repulsive, but there's *cis*.

  <img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/1_3.png" alt="1_1" style="zoom: 15%;" />

- For others, *trans* is almost always favored (8 kJ/mol).

  <img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/1_4.png" alt="1_1" style="zoom: 15%;" />

- However, some X-Pro structures are *cis*

##### $\phi$ and $\psi$ angles, the Ramachandran Map/Plot

Assumption: No two atoms in a molecule can come any closer than the sum of their **van der Waals radii**. Those who disobeys it are called "disallowed conformations".

- The plot is similar for most amino acids.

  <img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/1_5.png" alt="1_5" style="zoom:33%;" />

  - There are semi-allowed regions, where the distances are slightly less than the sum of vdw radii. 
  - For a certain kind of protein, we can draw the plot according to statistical data. Those who are out of allowed region might be in linkers, turns or loops.

- two special AAs

  - The plot of Gly has a larger area and is centralsymmetric. Both are due to the R group is simply a small H atom.
  - $\phi$ of Pro is fixed to about 60° because of the fixed cyclopentane. $\psi$ is most favored at about -60° and 160°
  - Gly often interrupts $\alpha$ and $\beta$. Pro cannot exist in anti-parallel $\beta$ sheet. Instead, they are often located in turns and loops.

##### structure features of typical conformations

| conformations           | approximate $\phi$ and $\psi$ (°) | features                                                     | special structures                                           |
| ----------------------- | --------------------------------- | ------------------------------------------------------------ | ------------------------------------------------------------ |
| $\alpha$ helix          | -57, -47                          | 3.6 residues/turn, 100°/residue<br />CO of i th and NH of i+4 th AA form H bond<br />side chains protude radically outside<br /><font color="grey" size=2>helical wheel projection: projection along the axis of $\alpha$ helix. 18 AAs, 5 cycles. Maybe nonpolar AAs are located on the same side and polar AAs on the other side </font> | 3<sub>10<sub/> helix: at the termini of a regular $\alpha$ helix |
| anti-para $\beta$ sheet | -139, 135                         | CO and NH on adjacent chains form H bond with each other to stablize | parallel $\beta$ sheet: a little more  tortuous H bonds      |
| $\beta$ turn            | --                                | i+1 is usually Pro and i+2 usually Gly<br />at the surface of protein (reverse the direction)<br /><font color="grey" size=2>nucleation center of folding</font> |                                                              |

#### side chain dihedral angles

- $\chi_1: \ce{N-C_\alpha-C_\beta-C_\gamma}$, $\chi_2: \ce{C_\alpha-C_\beta-C_\gamma-C_\delta}$, then goes along the side chain.

- Possiblities of $\chi_1$. All are avoiding overlaping with the main chain. Because $\ce{C=O}$ is the bulkiest group, *gauche^+^* ($\chi_1=-60°$) is the most favored. (trans is 2nd where $\chi_1=180°$)

  ![1_6](https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/1_6.png)

- $\chi$ tend to adopt staggered conformations (交错式构象).

  - For aromatic residues, $\chi_2$ tends to be $\pm 90°$ to minimize close contact with main chain.
  - For residues who form H bond with the environment, the last $\chi$ (Asp, Asn: $\chi_2$; Glu: $\chi_3$) adopts a wide range since that bond always rotates.

  <font color="grey" size=2.5> For those like Tyr who have two ways on the chain, both sides have the same $\chi$ number. This is not so meaningful.</font>

### In addition

1. tertiary structure: combination of secondary structures
2. quaternary structure: specific assembly of folded subunits (3rd struct)
3. examples
   - kertain: coiled coil ($\alpha$)
   - silkworm silk fibers: $\beta$
   - collagen: left-handed triple helix (Pro)



## Week2-1 Interactions in Proteins

### Basic concepts

Internal energy (内能) involves:

- kinetic energy
- potential energy

Recall three thermodynamical laws.

> Life is but a interplay of weak forces (non-bonding interactions).
>
> | bond                                | energy ($\mathrm{kJ/mol}$)                  |
> | ----------------------------------- | ------------------------------------------- |
> | ionic ($\ce{COO-}$ and $\ce{NH3+}$) | 20~80                                       |
> | hydrogen                            | 4~50, depending on distance and orientation |
> | dipole ($\ce{CO}$ and $\ce{CO}$)                  | 5~10                                        |
> | vdw                                 | 0.5~3 per atom pair                         |
> | hydrophobic                         | 4~8 per non-polar groups                    |
>
> functions:
>
> - additional strength to stabilize (high level structure)
> - flexible, interact and perform functions

### general electrostatic interactions

#### ion-ion

atoms carrying charge

- equation
  $$
  V=\dfrac{Q_iQ_j}{4\pi\varepsilon_0\varepsilon_rr_{ij}}
  $$

  - $\varepsilon_r$ : relative dielectric constant
  - The force is weaker in water (larger $\varepsilon_r=78.5$) than in hydrophobic core (though hard to measure)/other organic solvents.
  
- salt effect

  $$
  I=\dfrac{1}{2}\sum c_iq_i^2\\
  D=\sqrt{\dfrac{\varepsilon_0\varepsilon_rk_BT}{2N_Ae^2I}}\\
  V=\dfrac{Q_iQ_j}{4\pi\varepsilon_0\varepsilon_r}\dfrac{\exp(-r_{ij}/D)}{r_{ij}}
  $$
  
  $I \uparrow,\ V\downarrow$. "Neutralizing charges"


  eg: N and C terminal, side chain (salt bridge)

#### ion-dipole

- ion with groups carrying <u>no formal charge</u>
- dipole: **negative $\rightarrow$ positive** point charge
- dipolar molecule: those with dipole moment ($\ce{H2O, CO}$)

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/2_1.png" alt="2_1" style="zoom: 15%;" />
$$
V=\dfrac{Q_i\delta}{4\pi\varepsilon_0\varepsilon_r}(\dfrac{1}{r_{i-}}-\dfrac{1}{r_{i+}})=\dfrac{Q_i\delta}{4\pi\varepsilon_0\varepsilon_r}\cdot\dfrac{r_{i+}-r_{i-}}{r_{i-}r_{i+}}\approx\dfrac{Q_i\mu}{4\pi\varepsilon_0\varepsilon_rr_{ij}^2}
$$
eg: disolve salt. $\Delta H=E_{lattice}-E_{hydration}$, hydration: ion with water. 

entropy may increase.

#### dipole-dipole

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/2_2.png" alt="2_2" style="zoom: 15%;" />
$$
V=\dfrac{1}{4\pi\varepsilon_0\varepsilon_r}\left[\dfrac{\mu_i\cdot\mu_j}{r_{ij}^3}-\dfrac{3(\mu_i\cdot r_{ij})\cdot(\mu_j\cdot r_{ij})}{r_{ij}^5}\right]
$$
The potential depends on the relative orientation. The system tends to lower the energy. 

eg: 

- d-d interaction: $\ce{H2O}>\ce{HCl}$
- interaction between $\alpha$ helices (accumulation of peptide bond polarity), important for attracting charged molecules and enhances reactions.

### van der Waals

here refers to dispersion forces

- features
  - temporary dipole, induced non-uniform e^-^ distribution
  - contact distance, L-J potential
- important for molecules both with and without permanent dipoles
- significant for large molecules

### hydrogen bond

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/2_5.png" alt="2_5" style="zoom: 10%;" />

- features
  - short range: $r_{bond}\ll r<R_{vdw}$
  - directionality: 120~180°
- affects secondary structure & recognition, but not dominating/determining folding/assembly
- only intramolecular H bonds in the interior of a protein are favorable in presence of competition from water

actually vdW and H bond are both special cases of dipole-dipole interaction

### hydrophobic interaction

Oil doesn't mix with water. The molecules attract themselves more than each other. Thus mixing water and oil is only a process where water molecules form a cage (more ordered, to strengthen interaction) rather than thorough mixing. 

Their attraction increases ($\Delta H<0$) but the deterministic factor is $\Delta S(\ce{H2O})<0$. To minimize the entropy decrease, hydrophobic parts tend to gather to make their volume lowest and water molecules to form a cage least.

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/2_6.jpg" alt="2_6" style="zoom:25%;" />

### Disulfide bonds

Fold first, then disulfide bonds!



## Week2-2 Nucleic Acid Structure

### component

- pentose (difference)
- base
- phosphate (phosphoester bond)

### conformation

- bases point inwards
- plots of torsion angle
  - main chain: extended (trans, 180°)
  - $\chi$: anti (0°)
  
  <img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/2_3.png" alt="2_3" style="zoom: 50%;" />

### secondary structure

#### double helix

| helix           | (B) DNA   | ($\alpha$) protein |
| --------------- | --------- | ------------------ |
| units per turn  | (10.5) 10 | 3.6                |
| degree per unit | 36°       | 100°               |
| rise per unit/Å | 3.4       | 1.5                |
| diameter        | 2.2 nm    |                    |

#### grooves

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/2_4.png" alt="2_4" style="zoom: 90%;" />

| grooves        | major    | minor |
| -------------- | -------- | ----- |
| part of base   | more     | less  |
| actual H bonds | fewer    | more  |
| bind           | proteins | drugs |

Most proteins perform non-sequence-specific binding, switching on/off gene expression.

### forces to stablize

#### hydrogen bond

Waston-Crick base pairs vs unusual pairs

fraction of GC $\to$ melting temperature. AT: local unwinding

#### stacking interaction

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/2_7-2.png" alt="2_7" style="zoom: 60%;" />

<center><font color="grey">GC superimposition</font></center>

is a conbination of 

- vdW, where 0.34 nm is the optimal distance
- dipole-dipole, same direction $\Leftrightarrow$ repel

bases on the next level rotate 36° as a balance

#### electrostatic interaction

negative charged (-1 per bp) phosphate backbone: repulsion, controls distance between two strands

- low-salt: (phosphates) try to be trans; bases farther, prefer single strand (PCR)
- high-salt: screen (掩蔽) them to form double strand DNA

DNA prefers to take helix to 1) form stacking 2) avoid repulsion

#### with water

Water forms H bond with bases, pentoses and phosphate groups. The major and minor grooves potentially form same amount of H bonds. But due to larger space, the major groove needs more water molecules. This is the cause of **entropy loss**. 

As a result, the minor groove forms **more H bonds** with water, i.e. more enthalpy gain. Thus, it can trap more water molecules (a cluster that may not form H bonds) and compensate the entropy loss. In contrast, the major groove only maintains one shallow layer of water.

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/2_8.png" alt="2_8" style="zoom:50%;" />

<center><font color="grey">spine of hydration</font></center>

### other

- tightness: Z form (left-handed, seq-specific)>A form (seq-specific/RNA, high salt)>B form (normal)
- RNA structure



## Week3-1 HW

1. interactions

	|              | protein                                                      | nucleic acid                                                 |
	| ------------ | ------------------------------------------------------------ | ------------------------------------------------------------ |
	| interactions | covalent bond (disulfide)<br />electrostatic (ion and dipole)<br />vdW<br />H bond<br />hydrophobic | covalent bond<br />H bond<br />stacking (vdW, hydrophobic)<br />electrostatic |
	| secondary    | **H bond**                                                   | **H bond** (stacking)                                        |
	| tertiary     | H bond, **hydrophobic**                                      | H bond, **stacking**                                         |
	
2. heat and salt?

   salt stablizes the backbone, pushing the phosphate groups nearer and thus harder to be denatured by heat.
   
   a multi-layer structure: <img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/2_9.png" alt="2_9" style="zoom: 10%;" />

## Week3-2 Structure of Cell Membrane

functions of cell membranes

- compartmentalization
- selective permeable
- communication, signal transduction
- organize activities, energy production

### lipids

#### types

- glycerophospholipids
- sphingolipids
- cholesterol

#### features

- **amphiphilic**
- different content in different species or organs

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/3_1.png" alt="3_1" style="zoom: 25%;" />

#### lipids on the membrane

1. assymetric distribution (in two layers)

   - negative charged phospholipids tend to point at cytosol while the neutral ones point at extracellular space

     <p><font color="grey" size=2.5> when cell undergoes apoptosis, POPS goes to the outer membrane because entropy increases.</font></p> 

2. random distribution (in the same layer)

   - lipids are randomly distributed, and swim.. (bulk phase)
   
     - **vdW forces are not strong enough to hold lipids together**
     - hydrophobic effect drives the lipid to assemble, so there must be water

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/3_2.png" alt="3_2" style="zoom: 15%;" />


#### lipid rafts

- sphingolipids have longer and straighter fatty acids 

  thus have bigger vdW forces to hold themselves together (transiently)

  - glycerophospholipids: about 13?

- cholesterol can hold more lipid molecules
  - it's also very "hydrophobic"
  - it interacts with fatty acids in both sides of the plane, and has bigger surface area.
  - its ring is a rigid structure that makes lipid rafts stable.

![3_3](https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/3_3.png)

They assembly into lipid rafts, which

- are thicker (longer) than other parts of the membrane;
- are more resistant to detergents
- accommodate and gather proteins for specific functions like singaling

![3_4](https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/3_4.png)

### proteins

ratio $\uparrow$, function $\uparrow$

types:

https://en.wikipedia.org/wiki/Membrane_protein for figures

https://www.expasy.org/resources/protscale hydrophobicity plot

hydrophobic R chains point outwards (maybe hydrophilic inwards)

- integral (整合)

  - one or multiple **transmembrane** segments
  - $\alpha$ helix or $\beta$ barrel
  - some are not trans-membrane

- peripheral (外周)

  - no covalent bonds, only **non-covalent** bonds

    - electrostatically---with polar head
    - terminal hydrophobic group---with bilayer core
    - or bound to an integral protein

    <p><font color="grey" size=2.5> either outside and inside the membrane</font></p>

- anchored (锚定)

  - **covalently** bond to the lipids
  - GPI-anchored proteins (G: glycosylphosphatidylinositol-linked)
    - protein---phosphoethanolamine---tetrasaccharide---inositol---
      - where lipase C functions
    - phosphate---diacylglycerol, which is a part of bilayer
  - function: enzyme, antigen, adhesion

![3_5](https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/3_5.jpg)

### micelles

https://en.wikibooks.org/wiki/Structural_Biochemistry/Lipids/Micelles

- as the concentration of detergent/lipid molecules rises (over Critical Micelle Concentration, CMC), they are no more a layer on the surface, but forms micelles (胶束).

- structure

  - SDS can be seen as a cone (圆锥) model

    <p><font color="grey" size=2.5> which is due to electrostatic repulsion on the heads </font></p>

    - driven by hydrophobic interaction, they form a sphere
    - when concentration get higher, the sphere get bigger and water enter inside, thus hydrophobic interaction is disrupted.

  - DPC is a zwitterion but has to be parallel (replusion direction) to avoid contacting hydrophobic groups.

  - triglyceride may form a cylinder, which favours the formation of bilayer.

![3_6](https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/3_6.jpg)

<center>structure of molecules</center>

![3_7](https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/3_7.jpg)

<center>structure of SDS micelles</center>

### lipid mobility

- <font size=2.95>the more lipids are ordered/the closer lipids are placed, fluidity $\downarrow$, melting point $\uparrow$</font>

- Bilayer has two phases. gel phase: solid; fluid phase: liquid.

- saturated FAs adopt all-trans to achieve the closest contact, but all bonds are freely rotatable, especially those which are near the center of bilayer. 

  - gauche--trans--gauche makes a kink
  - cis-double bond makes a bend

  which reduces the packing density

  ![3_8](https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/3_8.png)

- cholesterol

  - block motions
  - disrupt ordered structure

  which makes a <u>balance</u>, modulating the fluidity. 

  ![3_9](https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/3_9.jpg)

  High conc cholesterol may abolish phase transition, keeping the membrane of eg. heart cells in cold blood animals functioning at a low tempearture.

- other motions

  - lateral (侧向): much faster in fluid phase
  - flip-flop: very slow



## Week4-1 Membrane Equilibrium

### terms

- extensive property: depend on the size or amount. additive

  - intensive property: e.g. density. 

- for intensive properties, define partial molar quantity: $\overline{Y_i}=\left(\frac{\partial Y}{\partial n_i}\right)_{T,P,n_j}$

  - then we can apply the additive rule.

- chemical potential: PMQ of Gibbs free energy
  $$
  \mu_i=\mu_i^0+RT\ln a_i
  $$

### chemical potential equlibrium

- for an open system (where $n$ may vary but $\mu$ doesn't vary under constant $T,P$)
  $$
  \mathrm{d}G=\sum_i \mu_i\mathrm{d} n_i
  $$

- for a multiphase system <u>in equilibrium</u>, which is separated by semi-permeable membrane

  - for each component $\mu_i$ should be the same and constant in all systems 
  - only when this component is permeable. ($\mathrm{d} n_i\neq 0$) 
  - <font color="grey">when not in equilibrium, $\mu$ may change first. But we only calculate about the equilibrium state </font>

  ![4_1](https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/4_1.png)

  <center>dialysis equilibrium</center>

- when equlibrium is broken

  If only one side ($\alpha$) has solute A of molar concentration $C$ that cannot pass through semi-permeable membrane. On the other side ($\beta$) is pure water with the same $T$ and $P$ (pressure of amtosphere on the liquid).

  <img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/4_2.png" alt="4_2" style="zoom:12%;" />
  
  - equilibrium can never be reached. **Osmotic pressure** is generated. 
  
  - Imagine we change the pressure to achieve equlibrium, with approximation, we can get
    $$
    \pi=P^\alpha-P^\beta=RTC
    $$
    
  - the solute can be either marcomolecules or small molecules

### Donnan effect

the ion may start transfering, but have to keep electrical neutrality.

Donnan effect: when one side contains impermeable charged molecule, concentrations of ions A,B in different side are different.

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/4_3.png" alt="4_3" style="zoom:15%;" />

Write the chemical potential
$$
\mu_1=\mu_{\ce{A}}^0+\mu_{\ce{B}}^0+RT\ln C_{\ce{A}}^\alpha+RT\ln C_{\ce{B}}^\alpha\\
\mu_2=\mu_{\ce{A}}^0+\mu_{\ce{B}}^0+RT\ln C_{\ce{A}}^\beta+RT\ln C_{\ce{B}}^\beta\\
$$
At equlibrium, $\mu_1=\mu_2$
$$
C^\alpha_{A}\cdot C^\alpha_{B}=C^\beta_{A}\cdot C^\beta_{B}
$$
Note $z$ is the charge and $C_M$ is concentration of the marcomolecule. According to electrical neutrality, let
$$
C^\alpha_{A}+zC_M=C^\alpha_{B}=a\\
C^\beta_{A}= C^\beta_{B}=b\\
r=\dfrac{zC_M}{2C^\beta_{A}}
$$
then solve it
$$
r_D=\dfrac{C^\beta_{A}}{C^\alpha_{A}}=\dfrac{C^\alpha_{B}}{C^\beta_{B}}=r+\left(r^2+1\right)^{0.5}
$$

If A is $\ce{H+}$, then $\mathrm{pH}^\alpha-\mathrm{pH}^\beta=\log r_D$

### combined effect: membrane potential

Consider this situation. $\ce{Na+}$ won't pass through because of the electric potential. Chemical potential equilibrium can never be reached, as in the example about osmotic pressure. This system looks like <u>the cell</u> where no ions can freely pass through.

![4_4](https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/4_4.png)

Part two of this section refers to the effect of different concentration, while part three refers to the effect of charged molecules.

Define electrochemical potential as their combination
$$
\mu^\prime=\mu+Zf\phi
$$

where $Z$ is charge, $f$ is Faraday constant, $\phi$ is the relative electric potential.

At equlibrium, for single ion A on one side:
$$
\Delta\varphi=\varphi^{in}-\varphi^{out}=-\dfrac{RT}{Zf}\ln\dfrac{c^{in}}{c^{out}}
$$
Ionic concentration difference and membrane potential are generated simultanously. Maintaining a concentration difference, we get membrane potential. Still, those which cannot pass through the membrane don't contribute.

For multiple ions ([reference](https://en.wikipedia.org/wiki/Goldman_equation)):
$$
\Delta\varphi=\varphi^{in}-\varphi^{out}=-\dfrac{RT}{Zf}\ln\dfrac{\sum\limits_{positive} p_ic_{i}^{in}+\sum\limits_{negative} p_jc_{j}^{out}}{\sum\limits_{positive} p_ic_{i}^{out}+\sum\limits_{negative} p_jc_{j}^{in}}
$$
where $p_i$ is permeability. The cell mainly regulates permeability and sometimes concentration.

> summary: 
>
> By putting charged marcomolecules and selectively letting ions pass through, the cell sets up the difference of concentration of ions as well as the electrical potential, acting as the fundation of activities like neural signaling.

### in addition

inference of osmotic pressure
$$
\mu_1^\alpha=\mu_1^0(T,P^\alpha)+RT\ln c_1^\alpha y_1^\alpha\\
\mu_2^\beta=\mu_2^0(T,P^\beta)+RT\ln c_2^\beta y_2^\beta
$$
where $c$ is the molar concentration of water. Here it's approximately the molar ratio (?)

Due to $G=U+PV-TS$, $\mu=V_0P$, $V_0$ is molar volume of water. 

When equilibrium, $\mu_1^\alpha=\mu_2^\beta$. Assume $y=1$ So
$$
\mu_1^0(T,P^\alpha)-\mu_2^0(T,P^\beta)=V_0(P^\alpha-P^\beta)=-RT\ln\dfrac{c_1^\alpha}{c_2^\beta}
$$




## Week4-2 HW

1. Why do most transmembrane helices contain more than the minimum number of residues?

   Prof: Usually helices form other H bonds with water except for the $\ce{C=O...H-N}$. But now the terminal few residues have to protrude out of the bilayer and form H bond with water to stabilize if the head group cann't.

   other reasons might be:

   - the helices are not perpendicular to the bilayer...
   - a few residues are used to anchor in the bilayer...

2. function of cholesterol: 

   - the formation of lipid rafts
   - modulate the fluidity
   
3. calculation

   try to calculate transfer



## Week5-1 Membrane Transport

### ionic composition


|               | main cation | main anion                   |
| ------------- | ----------- | ---------------------------- |
| intracellular | K           | organic phosphates and acids |
| intercellular | Na          | Cl                           |
| blood plasma  | Na          | Cl                           |

They are related to diseases. [reference here](http://book.bionumbers.org/what-are-the-concentrations-of-different-ions-in-cells)

### Principle 

- mostly decided by its <font color=\#8A2BE2>soluablity in oil</font>
- also highly depends on the size

#### diffusion through a protein-free lipid bilayer

[reference here](https://www.ncbi.nlm.nih.gov/books/NBK26815/)

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/5_0.png" alt="5_0" style="zoom:30%;" />

- can diffuse: 
  - gas: freely pass
  - hydrophobic: able to diffuse
  - small polar: much slower
- cannot diffuse: ions (charged) and large polar molecules (like zwitterions)
  - ions: hydrated, making it less soluable
  - also: may be attracted by the head group

#### classification

- mediated and non-mediated (gas, steriods...)
- passive and active

### mediated transport 

ionophores (离子载体): small molecules dissolved in the bilayer

#### carrier

##### action mechanisms

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/5_3.png" alt="5_3" style="zoom: 33%;" />

- reversible conformation change. 
- the direction is determined by the direction of gradient
- <font color=red>must just fit the site! </font>
- may need energy (light; ATP; other "downhill" potential)

##### classes

- uniport: only one, driven by gradient
- symport: one substrate driven by the gradient of another
- antiport: another substrate in reverse direction

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/5_1.png" alt="5_1" style="zoom:33%;" />

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/5_11.png" alt="5_11" style="zoom: 15%;" />

examples

- valinomycin (缬氨霉素) carrying $\ce{K+}$
  - takes the ion and move to the other side (size-specific)
  - can only bind to $\ce{K+}$ rather than $\ce{Na+}$
- lactose permease

#### channel

examples

- gramicidin (短杆菌肽): its dimer helices allows $\ce{K+}$ and $\ce{Na+}$ to pass through

- $\ce{K+}$ channel:

  ![5_2](https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/5_2.png)

  - acidic residues: attracting $\ce{K+}$ and exclude anions
  - $\ce{C=O}$ groups: coordinate with $\ce{K+}$ and replace its water molecules (cannot for $\ce{Na+}$)
  - the first one pushes the second out

- aquaporins

  - about 2.8 angstroms the narrowest, only allowing water molecules to passage



Spectroscopic Tools

```
categories

- electromagnatic wave
  - scatter
    - static
    - dynamic
  - absorption
    - UV/Vis
    - circular dichroism
    - IR
  - fluorescence
    - excitation
    - emission
    - anisotropy
- nucleus
  - NMR
```

## Week7-1 Circular Dichroism

- [Circular Dichroism (CD) spectroscopy - YouTube](https://www.youtube.com/watch?v=F8VGbgi1lwQ)

### Light and Polarization

is a kind of electromagnetic wave with mutually perpendicular oscillating electric and magnetic field.

Natural light is un polarized, i.e. ocscillating at all directions.

Two linearly polarized lights combine into linear/circular/elliptical polarized light.

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/7_1.png" alt="7_1" style="zoom:33%;" />

### Principle of CD

Linear polarized light can be seen as a combination of two circular polarized lights with different direction but same amplitude and initial phase.

Optically active substances (containing chiral center) has different absorbance to right/left-handed light (change $A$ and $\phi$). The emission light would be elliptically polarized light. 

> **Ellipticity** (椭圆度)
> $$
> \theta=\arctan \dfrac{A_{min}}{A_{max}}
> $$
> where $A_{min}$ and $A_{max}$ is the minimum and maximum amplitude of **the difference** (also elliptical) of two emission lights. Unit is degree.
>
> **Molar Ellipticity** ($\mathrm{deg\cdot cm^2/dmol}$)
> $$
> [\theta]=100\theta/cL
> $$
> where $c$ (mol/L) is concentration and $L$ (cm) is light path length.

notes on experiments:

- usually for 0.1 mM DNA/protein in 1cm cuvette, $\theta$ is 0.01~0.1 degree. usually use 0.2~0.5 mg/mL protein
- volume: 20 $\mu$L~1 mL; buffer: 10mM phosphate (lower interference)
- $\ce{Cl-}$ has strong absorbance at wavelength < 200 nm

features:

- CD occurs only when $\varepsilon\neq0$ (molar extinction coefficient).
- CD can be either positive or negative
- More averaged structure (AGCT; denatured) causes less intense signal; more asymmetry more intense

### Application

- determine protein/DNA <u>overall secondary structure</u>, study their <u>conformational change</u> as well as kinetics
- study the effect of ligand <u>binding, mutation</u> of protein. study the folding and denaturation

#### CD of DNA

features

- high salt B form: almost 0 at 280 nm
- no water A form: intense positive at 190 nm

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/7_2.png" alt="7_2" style="zoom:20%;" />

study conformational change: change temperature or water/organic solvent, plot spectrums together.

DNA SS is not so interesting as that of RNA/protein.

#### CD of protein

##### regions

- far-UV: 180~250 nm, 2° structures
- near-UV: 250(230?)~300 nm, 3° structure, if Phe/Tyr/Trp is in asymmetric environment (often folded structure)

##### standard curve

eg. poly-Lys in different environment

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/7_3.png" alt="7_3" style="zoom:25%;" />

| SS type        | maximum        | minimum          |
| -------------- | -------------- | ---------------- |
| $\alpha$-helix | 193 nm         | 222,208 nm       |
| $\beta$-sheets | 198 nm         | 213 nm           |
| random         | almost 0 > 220 | negative 190~200 |

#### application in protein study

##### determine SS fraction

For any protein, use 
$$
\Theta=f_\alpha\Theta_\alpha+f_\beta\Theta_\beta+f_u\Theta_u
$$
to fit the $\Theta-\lambda$ curve to find the parameters, i.e. the content of all SS.

- results are good for $\alpha$-helix and tolerable for the other two (only qualititively)
- may also use combination of other proteins with known content as basis (extract standard curve from them)

##### measure structural properties under various conditions

- change temperature/ligand concentration/pH/organic solvent/substrate
- record the change of a typical wavelength
- fit the curve to determine $T_m$, $K_d$, etc.

> - usually the one where <u>the biggest difference</u> between native and final state occured. 
> - For $\alpha$-helix, can pick 222 nm, often avoid 200 nm

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/7_4.png" alt="7_4" style="zoom:25%;" />

##### determine the effect of mutation

just compare the CD of WT and mutant

- if there's small change, but protein's function is affected, then this residue is responsible for the function
- if there's large change, then this residue affects the overall structure. We know it's critical for the structure but cannot determine if the residue is really responsible for the function (or at active site)

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/7_5.png" alt="7_5" style="zoom:15%;" />

##### study protein folding dynamics/kinetics

mix protein and denaturant, CD spectrum is recorded quickly. can get a $[\theta]$-time curve (at picked wavelength). may fit with exponential equation

> summary of plots
>
> - CD spectra ($\theta$-$\lambda$), SS and mutant
> - $[\theta]$-experiment condition
> - $[\theta]$-time



## Week7-2 Fluorescence Spectroscopy

- [Fluorescence Polarization - YouTube](https://www.youtube.com/watch?v=OdBNVrPvJMY)
- [Fluorescence Spectroscopy Tutorial - Typical Applications - YouTube](https://www.youtube.com/watch?v=MONv05mJu8Y)
- [slides in USTC](http://pic.ustc.edu.cn/_upload/article/files/40/4b/b8d018d3435b9990270250f1af43/b6481175-8dc0-4533-a210-b16fdddd6f2f.pdf)

### Principle

#### generation

![7-2_1](https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/7-2_1.png)

- electron has multiple energy levels
  
  - only when the input energy equals energy difference can an electron be excited
- vibration makes each energy state split
  
  - at equilibrium distance the energy is lowest
  - there is zero point energy
- fluorescence
  - only consider singlet states
  
  - range of spectrums
  
    - emission: **from ground virberational state in the first excited state (actually S'<sub>1<sub/>) to (any) ground state**
    - absorption: from ground virberational state in the ground state to any higher electronic energy state
      - thus absorption energy might be higher but little difference
  
  - features of the spectra
  
    - Stokes shift
    
      - energy of the first excited state might be lowered because of **the solvent**, etc., which makes the difference in absorption and emission spectrum.
    - shape of the emission spectrum is independent of absorption
      - they may intercept a little
      - and form kind of mirror symmetry
    - but the intensity is corresponding to the absorption
      - we'd choose the most absorbed wave length (intrinsic property)

#### properties of fluorescence

intensity is dependent on

- molar extinction coefficient (摩尔消光系数)
- fluorescence quantum yield
- concentration
- optical pathlength
- collect efficiency

considerations

- usually use relative intensity
- absorbance might turn nonlinear when concentration increases

fluorophores

- intrinsic: **Trp** >>Tyr
- nucleic acids' fluorescence is so weak
- small molecule to help: ANS, fluorescein, EB

characteristics: <u>conjugated C=C, complex rings</u>

#### fluorescence polarization

Situation: adding probes into the marcomolecule. Probes:

- without target or in aqueous solution, their fluoresence is quenched
- their fluoresence is enhanced in non-polar or rigid environment (eg DNA)

Use <u>horizontally</u> polarized light to excite, and detect horizontal and vertical signals (all are perpendicular to the exciting light)
$$
r=\dfrac{I_H-I_V}{I_H+2I_V}
$$
This $r$ actually measures **anisotropy** of the system (containing ligand).

- only molecules on certain orientation will be excited.
- at excited state (~nanoseconds) the molecules move randomly, thus the emission light is on all directions ($I_H\approx I_V$), making $r$ low (for single small molecule dye)
- the rotation is slowed (large $r$) if 

  - the probe binds to a big molecule
  - solution with high viscosity

![7-2_2](https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/7-2_2.png)

### Application examples

> In summary, the key points are
>
> - $r$ value is affected by molecular weight
> - from hydrophobic to exposed environment, red shift occurs and intensity probably decreases
> - figures
>   - intensity/$r-\lambda$ curve
>   - intensity/$r-$condition (mole,...but **usually no temperature**)

#### protein-protein/peptide/ligand

- One of the two has a flurophore and the binding will effect that site.
  
- $r-\lambda$ curve
  
  - bind to **cover Trp** into hydrophobic site$\longrightarrow$blue shift
  - hard to predict intensity
  
  <img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/7-2_4.png" alt="7-2_4" style="zoom:50%;" />
  
- $r-$molar ratio
  - is binding? see shape
  - how many? see ratio (there must be apparent mw change)
  - binding mode (close to Trp)? see $r$
  - affinity??

#### conformational change

eg: add urea to protein (with **Trp**!)

- slight red shift
- intensity significantly reduced

make a plot of fold fraction (calculated from intensity) and urea concentration to study denaturation/folding/Kd...

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/7_2-3.jpg" alt="7_2-3" style="zoom:35%;" />

#### introduce to living cell

Design these proteins to detect the content of small molecules, like FICRhR/cAMP. 

As a technique to study the change/visualize processes ...

### Fluorescence Resonance Energy Transfer

#### principle

FRET is a kind of dipole-dipole interaction beween excited states. The donor is excited (eg by light) and then excites the acceptor. eg: CFP and YFP

The efficiency of energy transfer is $\dfrac{R_0^6}{R_0^6+r^6}$, which decreases quickly through distance.

> Primary conditions:
>
> - donor and acceptor is close contact (10~100 $\AA$)
> - the absorption spectrum of acceptor **overlap**s with the emission spectrum of the donor (but separable).
> - their dipole orientation must be approximately parallel

#### application

Strength: just fusion expression of a fluorescent protein, easier than chemical dye in processing and maintaining biological activity (stability).

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/7-2_5.png" alt="7-2_5" style="zoom:20%;" />

Can quantify:

- Protein-Protein/DNA Interaction
  - if they can interact, you'll see the emission spectrum of the acceptor (predominant) and donor, instead of only the donor's.
  - *in vivo* or *vitro*
- protein conformational changes
  - in a specifically designed conformation, the donor and acceptor can interact; condition changes, spectrum changes
  - maybe observe signal to change with time, if slow

https://www.biomart.cn/mobile/experiment/article/25123.htm

### Quencher

Molecules which can interact with others (fluorophore) and reduce the intensity of fluorescence. 

eg: small molecules as quenchers. like oxygen, acrylamide, $\ce{Cs+*4H2O}$

Put (different concentration of) quencher together with fluorescent protein (marcomolecule) to measure the **accessibility** for the quencher into the <u>aromatic residues</u>. Just measure $F_0/F_c$, the intensity before and after adding quencher.



## Week8-2 NMR

- [Relaxation (NMR) - Wikipedia](https://en.wikipedia.org/wiki/Relaxation_(NMR))
- [How MRI Works - Part 1 - NMR Basics - YouTube](https://www.youtube.com/watch?v=TQegSF4ZiIQ) detailed
  - [NMR spectroscopy visualized - YouTube](https://www.youtube.com/watch?v=RZLew6Ff-JE)
  - [eLearning - NMR Relaxation - YouTube](https://www.youtube.com/watch?v=DQ5kO-jPQyA)

### Principles

- static magnetic field: make the energy difference of nucleus

  - spin energy levels split, because spin has direction

- inducing electromagnetic wave: to excite the nucleus when the energy <u>matches</u>
  $$
  \Delta E=h\nu=h\gamma B_0
  $$

  - $\nu$ is the frequency of inducing electromagnetic wave
  - $\gamma$ is only related to atom type
  - $B_0$ is the strength of static megnetic field

- the absorption and emission energy is the same, so we often measure the latter

- for ^1^H, 42.58 MHz~1 Tesla

### Phenomenons

#### Chemical shift

##### Definition

ring current of electron $\longrightarrow$ induced magnetic field $\longrightarrow$ $B$ on the nucleus changes

$$
B_{eff}=B_0(1-\sigma)
$$

so

$$
\nu=\gamma B_{eff}\\
\delta=\dfrac{\nu-\nu_{ref}}{\nu_{ref}}\times 10^{6}
$$

$\nu_{ref}$ is from a standard substance.

##### Factors influencing $\delta$

> The induced magnetic field is related to
>
> - Electron density: more electron (more shield), higher current, strong field, lower $\delta$
> - the direction (spatial orientation) of other atoms

so the factors include:

1. inductive effect (electronegativity)

   connecting to such an atom lowers the electron density ($\ce{OH,C=O,C+}$...)

   thus farther from TMS i.e. large $\delta$

   roughly: $\delta \propto $ Electronegativity of connected atom

2. hybridization effect (anisotropy)

   usually vinyl Hs have big $\delta$

   for biomolecules, those on the aryl (F, W, Y)

3. other effects (structure)

   - H bond
   - dihedral angle, secondary structure
   - temperature
     - random coil is more variable
   - solvent

#### J coupling (spin-spin coupling)

For two protons on two connected C, when one of them is on ground state/excited spin state, it generates different induced magnetic field (only differ in directions). It functions on the other and makes the $\Delta E$ change. The magnetic field in one direction strengthens $B_0$ while the other weakens, so $\Delta E$ splits into two whose energy difference is $J$. This effect is mutual.

- often < distance of 3 bonds.
  - <u>transmitted via electrons in the intervening bonds</u>
- NMR intensity: number of Hs; split: number of adjacent Hs 
  - just add up the effect of all Hs, 2 choices each
  - get probability distribution/strength: Pascal triangle

#### Relaxation

Relaxation time $T_2$ describes how fast the NMR signal decays in time-domain spectroscopy. 

In frequency-domain spectroscopy, full width at half maximum (FWHM, linewidth) is $\pi/T_2$ (The AUC is constant). Broad peak/large FWHM means 

- small $T_2$/slower decay, lower height
- molecule size is big, or this group is rigid (like -Ph)

#### Labile hydrogens

H on O,N,-COOH are shielded by hydrogen-deuterium exchange (add D<sub>2<sub/>O)

### Applications

only 1D ^1^HNMR

#### identify inhibitors

eg: Cyclophilin catalyzes isomerization of Pro in the substrate polypeptide from *cis* to *trans*. *cis* exists in single substrate, disappears when adding Cyclophilin, re-appears when further adding its inhibitor.

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/8_1.png" alt="8_1" style="zoom:15%;" />

#### study protein folding

- signals of unfolded protein are simular and overlap
- that of folded protein have dispersion, and may have signals <0.5 (alkyl) or >9 (peptide bond) ppm.

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/8_2.png" alt="8_2" style="zoom:25%;" />

#### drug discovery

Compare signals of <u>small molecules</u>. The signals may look like the target after binding. eg: They move together, so peaks of ligands become broader

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/8_3.png" alt="8_3" style="zoom: 20%;" />

#### metabonomics

First we'd know the characteristic peak of compounds.

##### biomarker identification

Compare the spectra of samples (like urine) from patients and healthy ones. Find the abnormal metabolite corresponding to abnormal signals.

##### toxicology

Compare the spectra of samples from rat treated with a drug/toxin and control. Identify which is up/downregulated.

![8_4](https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/8_4.png)



### Summary on biophysical techniques

- CD tells about SS, but don't know if it's properly folded
- fluoresence/NMR can tell if it's folded, but cannot tell exactly what SS is

<font color=grey>having $\alpha$ helix does not mean it's properly folded/packed! </font> 



## Week10-1 Conformational Transition

### Fundamental Representations

For the transition of a marcomolecule in 2 conformations $\ce{A<=>B}$, the equilibrium constant is $K=\dfrac{[B]}{[A]}$, the probability of observing B is 
$$
P_B=\dfrac{[B]}{[A]+[B]}=\dfrac{K}{1+K}
$$
Assume there are many states ($i=1,2,\dots$) in equilibrium, and a supposed reference state ($i=0$, only for the sake of mathematics), define **weight** as the <u>equilibrium constant</u> between states (S):
$$
\omega_i=\dfrac{[S_i]}{[S_0]}=\exp(-\Delta G_{i,0}/RT)
$$
so $\omega_i$ is only related to energy level. We combine those in the same energy level, the number of states is called **degeneracy** $g_i$. So we <u>redefine</u>: $i$ does not refer to a single state but the combination of states in the same energy level, so
$$
\omega_i=g_i\exp(-\Delta G_{i,0}/RT)
$$
We also define **partition function** as
$$
Q=\sum_{i=1}^{N_{tot}} \omega_i
$$
and the probability will be 
$$
P_i=\dfrac{\omega_i}{Q}
$$
<font color="grey">Sum of concentrations on this energy level divided by the total concentration. All divided by $[S_0]$ and sum up states in the same energy level to get $\omega_i$.</font>

### Conformational Transition Models

Notation:

- For a DNA/protein of length $N$, each unit has a state (helix or coil, etc.), represented as a,b. 
- Suppose $\Delta G$ of changing a single unit from a to b is $s$. 
  - so $\omega_i=g_is^i$
- Probability of the state with $i$ units in b state is named $P_i$.
- Fraction of a,b in all units of all states is $P_a,P_b,\dots$
  - if define fraction of b in state $i$ as $f_i=\dfrac{i}{N}$,
  - then $P_b=\frac{f_i\omega_i}{Q}$

#### all-or-none (highly-cooperative) model

<u>Only two states</u>: all is a, and all is b.
$$
K=s^N\\
P_0=P_a=\dfrac{1}{1+s^N}\\
P_N=P_b=\dfrac{s^N}{1+s^N}\\
$$
Interpretation: 

- If $s=1$ (no energy difference), $P_0=P_N=1/2$

- When $N$ is large,

  - as condition (pH, temp) change, $s$ changes

  - as long as $s>1$, $P_b$ is close to 1; if $s<1$, $P_b$ is close to 0

  - which means $P_b$ will not stay at the middle but either close 0 or 1

  - the graph looks like $\downarrow$ , where it seems the two states change **suddenly**

    <img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/10_1.png" alt="10_1" style="zoom:15%;" />

#### non-cooperative model

<u>All units change independently.</u> So there will be $N+1$ states and
$$
g_i=\binom{i}{N}=\dfrac{N!}{i!(N-i)!}\\
Q=\sum \binom{i}{N}s^i=(1+s)^N\\
$$
Interpretation

- if $s=1$, $P_i$ will be a binomial distribution; otherwise just modified by $s^N$

- $P_b=\cdots=\dfrac{s}{s+1}$, and this might be a long dependence

  <img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/10_2.png" alt="10_2" style="zoom:15%;" />

- $P_b/P_a=s$ which is independent of $N$. The fraction of b in all states is the same as single a and b.

#### Zipper Model

A classical model to describe realistic transitions. 

- Assume it's impossible for two sites to transition simultanously.

- Assume the first step is more difficult (**nucleation**, <font color=grey>the one to break original complete structure, see examples later)</font>, which needs extra energy. 
  $$
  \sigma=\Delta G_{extra}>0\\
  \omega_1=\exp[-(s+\sigma)/RT]
  $$
  
- Also, only those next to the nucleation center will perform transition. <u>State $i$ will have continuous b units of number $i$</u>. 

So
$$
g_i=
\begin{cases}
1 & i=0\\
N-i+1 & i\geqslant 1
\end{cases}\\

\omega_i=
\begin{cases}
1 & i=0\\
g_i\sigma s^i & i\geqslant 1
\end{cases}\\

P_b=\dfrac{\sum\limits_{i=1}^{N}\dfrac{i}{N}(N-i+1)\sigma s^i}{1+\sum\limits_{i=1}^{N}(N-i+1)\sigma s^i}
$$
Results (try to simulate yourself!):

- it will look like highly-cooperative model if

  - $\sigma \downarrow$ ($\ll s$), difficult for the first to occur. 
  - $N \uparrow$, $\sigma$ is more distributed on the sequence

  so $\sigma$ is called "cooperativity coefficient"

- (not like in slides) it will not look like non-cooperative model because definition of degeneracy has implied cooperativity.

![10_2_3](https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/10_2_3.png)

### Applications

#### in proteins

##### coil-helix transition

From coil to helix, the first 4 (/3/5) events are difficult. Then just need to move one residue if the neighboring three are ready.

- $s\approx1$ because enthalpy gain from H bonding is counteracted by entropy loss (helix is more fixed)
  - <font color=grey>sequence dependence: AA type and its environment</font>
- $\sigma\approx2\times10^{-4}$ for poly-$\gamma$-benzyl-L-Glu
  - Accounting for entropy loss from 3 more residues (N just minus 3). Thus, $\sigma_{3-10}>\sigma_{\alpha}$ (i-i+3)
  - <font color=grey>sequence dependence: Lys $\sigma\approx10^{-3}$</font>

##### helix melting (the reverse process)

Melting one residue may start from the end or middle. The latter situation will break H bonds at both ends. <font color=grey>Maybe break 5 H bonds and gain entropy once? A larger $\sigma$??? </font>

Still using the original $s$ which $>1$. 

- from the end: $\omega_{N-1}=2\sigma s^{N-1}$ 
- from the middle: $\omega_{N-1}=(N-2)\sigma^2 s^{N-1}$
  - <font color=grey>the first term is degeneracy. 2 ends, others are middle</font>
  - <font color=grey>if you consider the reversed numbering, $\omega_1=(N-2)\sigma^2\dfrac{1}{s}$, divided by $s^{N}$ </font>

> So when the probabilities equal,
> 
> $$
> N=\dfrac{2}{\sigma}+2
> $$
> 
> $N$ is independent of $s$ but only decided by $\sigma$


Larger $N$ distributes the nucleation loss and more likely to start in the middle. 

But more start from the end. eg: poly-$\gamma$-benzyl-L-Glu will melt in the middle if $N>10002$....

#### in DNA

It follows the same equations to determine whether to start from the end or middle. More probable in the middle? $\sigma\approx10^{-7}$...

##### DNA melting

From dsDNA to ssDNA, 

- $s$ is composed of energy to break (2/3) H bonds and a pair of base stacking (main force in DNA secondary structure!). 
- $\sigma$ means additional energy to break base stacking on the other side.

##### DNA annealing

- dependent on the concentration of ssDNA, so $\omega_1=\kappa\sigma s$
- sequence dependence
- <font color="grey">in annealing it needs finding proper combination of base pairs (eg. a primer searches for complementary sequence), so cooperativity is lower than melting. Also $T_m\gg T_a$. I think it's combining $\kappa\sigma$ as $\sigma$ and this is just amending non-quasi-static kinetic error. </font>

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/10_2+.png" alt="10_2+" style="zoom:25%;" />

##### DNA helical transition

In ethanol (remove water), $\ce{B-DNA->A-DNA}$.

- think of DNA becoming a little bit tighter.
- $s$ : the difference between average energy per base pair in two forms
- $\sigma$: A-B junction, it's a little distorted when an A form base connect with a B
  - 5~8 kJ/mol, $\sigma=0.14\sim0.04$

Rise salt conc/alcohol conc/temperature, $\ce{B-DNA->Z-DNA}$

- $\sigma$ turns to the energy used to turn right-handed helix into left-handed helix
- $\sigma$ is larger if conditions suitable for Z-DNA (above) is satisfied.

##### DNA supercoil state transition

For double-strand circular DNA

- linking number: how many times one strand comes around the other. 
  - Might change only when double strand is broken, like using topoisomerase
- twisting number: how many "turns" the two strands rotate along central axis.
  - total number of base pairs in a DNA molecule divided by the *number* of bases per turn
- writhing number: the number of supercoiling

$$
L=T+W\\
\Delta L=\Delta T+\Delta W
$$

situations

1. In DNA replication, helicase functions. $L$ remains. $T$ behind decreases, $T$ in front increases.
2. An enzyme unwinds DNA by 2 turns through cutting it off. $\Delta L=-2$
   - maintaining circular shape: $\Delta W=0$
   - relive all stress: $\Delta T=0$

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/10_3.png" alt="10_3" style="zoom:20%;" />

Transition occurs in a local region of alternating C and G

why supercoil?

- save space
- for replication and transcription (Z-DNA)



<font color="skyblue">In addition for DNA melting: how does $s$ change with condition?</font>

- at $T_m$, melting and renaturation reach a equilibrium, $\mathbf{s_{T_m}=\dfrac{[melted]}{[unmelted]}=1}$
- according to [Van 't Hoff equation](https://zh.wikipedia.org/wiki/范特霍夫方程), $s=-\dfrac{\Delta H^\Theta}{R}(\dfrac{1}{T}-\dfrac{1}{T_m})$
- when $T>T_m$, $s>1$ thus can melt; when $T<T_m$, $s<1$



## Week10-2 Protein Folding

### Basics

Why study protein folding?

1. sequence-structure relationship
2. related to diseases
3. biotechniques eg. purify

#### some facts

- Protein folds in milliseconds
- Protein folding must follow certain pathways
  - instead of randomly search in the huge conformation space.
- Protein refolds automatically upon dilution of denaturant.
- Unfolded states are heterogenous (all kinds) while native states are narrow.
- Typical denaturant: 8M urea, 6M GdnCl

#### concepts

1. denatured state (D)

   - an ensemble of dynamic inter-converting species

   - with few nonlocal interactions; only retain few local interactions

     (interactions between atoms that are close in sequence)

2. intermediate (I)

   - occur during refolding of large proteins
   - on-pathway intermediate has native-like interactions that makes conformation searching easier
   - trapped intermediate slows down folding, which needs chaprons, <font color=grey>otherwise aggregates/precipatates (even lower energy than native)</font>

3. transition state (TS)

   - highest energy, correponding to rate-limiting step
   - backbone: forms secondary structure
   - side chain: specific rotamers, a bridge from disordered conformation to native structure

### Folding Models

1. framework model
   - quickly form secondary structure first, leading to tertiary structure
   - segaments of native-like structures in TS, some residues have high $\phi$ value
   - prefered in bigger protein
   - subvisions: "diffusion collison" and "propagation"
2. nucleation model
   - form a compact nucleus first, then further condensation
   - partially formed interactions; residues have equalized $\phi$ value
   - prefered in smaller protein

### Equilibrium Studies

#### principles

states: D, N, Intermediates (know how many!)

choose a property $y$ (eg. Fluo/CD under a certain $\lambda$ that maximizes the difference). 

fit the fraction $f$ of each state.
$$
y=f_Ny_N+f_Dy_D+\sum_if_iy_i\\
f_N+f_D+\sum_if_i=1
$$
define the normalized property (distance from denatured)
$$
d_i=\dfrac{y_i-y_N}{y_D-y_N} \in [0,1]
$$
and several ratios
$$
K_{i}=\dfrac{f_{i}}{f_N}\\
K_{D}=\dfrac{f_D}{f_N}
$$
and apparent (表观) franctional denaturation/native
$$
f_{app}=f_D+\sum_if_id_i\\
1-f_{app}=f_N+\sum_if_i(1-d_i)
$$
so the apparent equilibrium constants (between apparent denatured and native)
$$
K_{app}=\dfrac{f_{app}}{1-f_{app}}=\dfrac{K_D+\sum\limits_iK_id_i}{1+\sum\limits_iK_i(1-d_i)}{}
$$

- if there's no intermediates (2 states), $K_i=0$

  condition: $K_{app}=K_D=-\exp(\Delta G/RT)$

  or is constant in different property experiments

- otherwise $K_{app}\neq K_D$, or varies in experiments since $d_i$ varies

#### experiment and fitting

add denaturant/refolding buffer; get signals under different conditions

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/10_4.png" alt="10_4" style="zoom:20%;" />

##### denaturant (GdnCl/urea, Fluo/CD)

(Assume $y_D>y_N$) 2 states, $f_D=\dfrac{[D]}{[D]+[N]}=\dfrac{K_D}{K_D+1}$. 

For $\ce{N<=>D}$, $\Delta G$ has approximately linear relationship with conc of denaturant $c$. Thus $K_D=\exp[-(\Delta G_0+mc)/RT]$, where $\Delta G_0$ is when $c=0$.

Plug in $y=y_N+(y_D-y_N)\cdot f_D$ (or reverse D/N), get
$$
y=y_N+(y_D-y_N)\cdot\dfrac{\exp[-(\Delta G_0+mc)/RT]}{\exp[-(\Delta G_0+mc)/RT]+1}
$$
Fit the y-c curve to get $m$ and $\Delta G_0$. Either set $y_D$, $y_N$ as parameters or not. 

##### temperature (CD)

2 states, still $y=y_N+(y_D-y_N)\cdot\dfrac{K_D}{K_D+1}$, but $K_D$ is related to $T$.

Fit curve to get

- $\Delta H_m$: at $T_m$
- $\Delta C_p$: of reaction. assume as constant
- $T_m$: D and N reach equilbrium
- initial protein concentration

### Kinetics Studies

#### time-scale experiment 

For 2-state model $\ce{N<=>[k_1][k_{-1}]D}$, $y=y_\infty+A\exp(-k_ut)$. Fit $\ln(y-y_\infty)-t$ to get $k_u$. Reversely get $k_f$.

for 3-state model $\ce{N<=>[k_1][k_{-1}]X<=>[k_2][k_{-2}]D}$, $y-y_0=A_1\exp(-\lambda_1 t)+A_2\exp(-\lambda_2 t)$. Maybe fit each subsection. 

Also, intermediate of folding and refolding may not be the same.

#### get rate for 2-state model

##### equations

For $\ce{N<=>D}$, do unfolding/refolding separately. 

- Add denaturant into folded protein
  - **when conc is high, it's only unfolding**, discard refolding reaction (vice versa)
- Repeat this 2-state experiment with different concentration. Get a $k-c$ curve.
  - Actually regard this as $\ce{N + urea ->[k_1] D + urea}$.
- Fit $\ln k_f=\ln k_{\ce{H2O}}-m_{k_f}c$.
  - The linearity indicates if it's 2-state.

Do the same reversely and combine the plots, the two arms.

Or just fit all data with this to get Chavron plot <font color=grey>(assume folding rate in high concentration is much slower than unfolding and vice versa).</font>
$$
k=k_{f,\ce{H2O}}\cdot\exp(-m_{k_f}c)+k_{u,\ce{H2O}}\cdot\exp(m_{k_u}c)
$$

##### Interpretation

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/10_5.png" alt="10_5" style="zoom:20%;" />

- All parameters >0.
  - $k_{\ce{H2O}}$ is the rate when $c=0$
  - $m_k$ is an important factor to indicate how denaturant changes the rates
- The intersect of two lines indicates the concentration when half the protein is unfolded, indicating thermodynamical stability.
  - $c_m=\dfrac{k_{f,\ce{H2O}}-k_{u,\ce{H2O}}}{m_{k_f}+m_{k_u}}$
- simple compare: kinetically more stable if
  - $m_{k_f}$ equals and $m_{k_u}$ smaller;
  - $m_{k_u}$ equals and $m_{k_f}$ bigger;
- $\ln k_f$ and $\Delta G_F$ are both linear with $c$, and the coefficient should be the same.
  - But if we use only one $m$ is equilibrium study, i.e treat it as 2-states, add up two $\Delta G$, then the total $m=m_{k_f}+m_{k_u}$. 
  - Larger $m$ leads to steeper equilibrium curve (like all-or-none), indicating more stable besides the above two criteria.

### Local State Studies

#### NMR

Collect the signal of all $\ce{-NH-}$ groups. All residues in HSQC have been assigned.

- Compare the ^1^H chemical shift of (half?) denatured protein and ramdom coil
  - whose chemical shift is lack of dispersion (8~8.5 ppm). If some residues are out of this (7.5~9.5 ppm), it may maintain folded structure.
- Compare the $\ce{H-N-C_\alpha-H_\alpha}$ J-coupling constant, which is related to $\phi$ angle.
- Check the NOEs.

???

#### transition state ($\phi$ value anaylsis)

For $\ce{N<=>TS<=>D}$, compare WT and one-residue mutant. 

Experimentally get $k_f$ and $k_u$ (<font color=grey>I guess under appropriate denaturant concentration, if we assume $m$ does not change. But they do change...</font>).

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/10_6.png" alt="10_6" style="zoom:40%;" />

Think of activation energy: $k_f=k_{f,0}\exp(-\Delta G_{U-TS}/RT)$. Also for $f_u$. 

Intuition: larger $k_f$, smaller $|\Delta G_{U-TS}|$. 

Strange definition:
$$
\Delta\Delta G_{U-TS}=\Delta G_{U-TS}-\Delta G^*_{U-TS}=-\ln\dfrac{k_f}{k_f^*}\\
\Delta\Delta G_{TS-N}=\Delta G^*_{TS-N}-\Delta G_{TS-N}=-\ln\dfrac{k_u^*}{k_u}
$$
then 
$$
\phi_f=\Delta\Delta G_{TS-N}/\Delta\Delta G_{U-N}\\
\phi_u=\Delta\Delta G_{U-TS}/\Delta\Delta G_{U-N}
$$

- for 2-state model, $\phi_f+\phi_u=1$
- if $\phi_f\approx1$, no difference in TS and U, so this residue is absent in TS formation; if $\phi_u\approx1$, no difference in $\Delta G_{TS-N}$, so this residue is almost done in TS
- if most residues shows 0 or 1, the protein is completing local structure first: **Framework model**
- if most residues shows fractional $\phi$, the protein is forming partial structure first overall: **Nucleation model**

Notes 

- $k_f$ is on the "unfold" side.
- $\phi_f$ is on the "fold" side.
- $K_u=\dfrac{k_u}{k_f}$ for $\ce{N<=>U}$, because $r_u\propto k_u[N]$

#### pulsed hydrogen labeling

- unfold the protein in $\ce{D2O}$, thus all $\ce{-NH-}$ carry D
- dilute with $\ce{D2O}$ buffer to make it partially refolded (for a while)
- dilute with excessive $\ce{H2O}$ (for a while) at high pH (not that many H)
  - those which are unfolded carry H.
  - those which folded before maintains D: analyze by 2D NMR
- finish folding by lowering pH



## Week12-1 Protein-Protein/Ligand Interaction

### Basics

- non-covalent interactions: hydrophobic, ionic, H bonds...
- binding sites: inside? narrow? 
  - small ligand: pockets in/outside
  - larger: surface (peptide) or deeply inside (heme)
  - huge: protein in grooves of DNA
- structure 
  - lock-and-key model: complementarity
  - structural changes to minimize energy (flexible)

### Qualitative Study

#### "pull-down" assay

[Pull-Down Assays | Thermo Fisher Scientific - SG](https://www.thermofisher.com/sg/en/home/life-science/protein-biology/protein-biology-learning-center/protein-biology-resource-library/pierce-protein-methods/pull-down-assays.html)

- *in vitro*
- fusion protein
- other combination

![12_1](https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/12_1.png)

#### NMR CSP

- *in vitro*
- residues that shift a lot are important

$\Delta \mathrm{ppm}$

#### yeast two hybrid

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/12_2.png" alt="12_2" style="zoom:50%;" />



#### Applications

- screening cDNA library to assign functions; identify drug target
- mutation study (residues involved in interaction/binding)
- drug discovery/find regulator compounds

Maybe we can compare the interaction strength by observing expression level (increase/decrease all matter)...

### Quantative Study

#### thermodynamics

For 1:1 binding
$$
\ce{M + L <=>[k_{on}][k_{off}] ML}\qquad
\Delta G=-RT\ln K_a=\Delta H-T\Delta S
$$
where $K_d=1/K_a$. ($[ML]=K_a[M][L]$; $[M][L]=K_d[ML]$)

We wish low $K_d$ and negative $\Delta G$

#### ITC

[Isothermal Titration Calorimetry (ITC) | Center for Macromolecular Interactions (harvard.edu)](https://cmi.hms.harvard.edu/isothermal-titration-calorimetry)

appratus:

<center><img src="../../course/molecular biophysics (NUS)/molecular biophysics.assets/ITC1.png" alt="ITC1" style="zoom:60%;" align=center /></center>

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/itc_sample_data.png" alt="itc_sample_data" style="zoom:60%;" />

Fit the curve, get $\Delta H$, $\Delta S$, $K_a$, $n$ (stochiometry, 化学计量)

#### equilibrium and saturation binding

##### single site

define fraction of M that has L on the site: 
$$
\nu=\dfrac{[ML]}{[ML]+{[M]}}=\dfrac{K_a[L]}{K_a[L]+1}=\dfrac{[L]}{K_d+[L]}
$$
assuming $[L]\gg[M]$ (L not consumed in binding, use $[L]_{total}$)

This will be a Michaelis-Menten-like curve.

##### multiple equivalent sites, no interaction

Just multiply by $n$ because they are identical and independent. Also, the effect is proportional to experimental measure. We also define binding fraction of all sites.
$$
\nu^*=\dfrac{nK_a[L]}{K_a[L]+1} \in[0,n)\\
Y=\dfrac{\nu^*}{n}=\dfrac{K_a[L]}{K_a[L]+1}\in[0,1)
$$
equivalently，
$$
\dfrac{\nu^*}{[L]}=K_an-K_a\nu^*
$$
this is called **Scatchard equation**. We can get $K_a$ (slope) and $n$ (x intercept).

##### multiple non-equivalent sites, no interaction

Due to independence, we can add up the results of all kinds of sites.
$$
\nu^*=\sum_i\dfrac{n_iK_{a,i}[L]}{K_{a,i}[L]+1}
$$
But the results given:

- there's $n$ separate lines, the slope of which is $K_{a,i}$
- We can still know $\sum\limits_i n_i$ from x intercept.

##### multiple non-equivalent sites, interacting

The Scatchard plot becomes a curve. Modify:
$$
\nu^*=\dfrac{nK_\pi[L]^{n_h}}{K_\pi[L]^{n_h}+1}\\
K_\pi=\prod_iK_i
$$
still $Y=\dfrac{\nu^*}{n}$, which can come from simple Scatchard plot...

define $\theta=Y/(1-Y)=K_\pi[L]^{n_h}$, then got Hill equation:
$$
\ln\theta=\ln K_\pi+n_h\ln\left[L\right]
$$
Interpretation:

- fit the equation to get $K_\pi$ (exp y-intercept) and $n_h$ (slope)

- if no interaction, $\ln\theta-\ln\lef[L]$ curve will be straight, or $n_h=1$

- if $n_h<1$, negative cooperativity, affinity $\downarrow$

- if interaction, the curve looks like

  <img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/12_3.png" alt="12_3" style="zoom: 25%;" /><img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/12_4.png" alt="12_4" style="zoom:25%;" />

  <font color=grey>At begining, too few, no cooperativity; in the end, no more sites. Thus slope=1?</font>

  extrapolate the first and last part of the line and get x-intercept: $-\ln K_{a,1}/n_{h,1}=-\ln K_{a,1}$, also $-\ln K_{a,n}$

  - if $n_h>1$, it looks like the left figure, $K_{a,1}<K_{a,n}$
  - if $n_h<1$, it looks like the right figure, $K_{a,1}>K_{a,n}$

#### CD and fluorescence

change $[L]$ and get different signal

- fluorescence: strength of maximum difference. 280 nm for Trp

- fluorescence polarization: use $r$ as $\nu^*$

  - rotation relaxation time: $p=\dfrac{3nV}{RT}$

    where $n$ is viscosity and $V$ is molecular volume

  - $r=\dfrac{I_H-I_V}{I_H+2I_V}$, and according to Perrin equation

    larger $V$, longer $p$, larger $r$

  - advantages

    - independent on light intensity, fluorophone conc, etc, not measuring intensity but the ratio between components
    - less affected by instrumental change

- CD: both far (2°) and near UV (3°)

- eg: single site saturation. fit $Y=\dfrac{B_{max}[L]}{K_d+[L]}$

  maybe: a linear term can be added to account for non-specific background binding

### Kinetics of Binding

#### experiments

##### surface plasmon resonance (SPR)

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/12_5.png" alt="12_5" style="zoom: 33%;" />

steps

- immobilization: fix protein/ligand to the surface covalently
- association: add ligand/protein
- equilibrium: measure association
- dissociation: measure
- regeneration: break the remaining complex by acid/base

##### biolayer interferometry

![12_7](https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/12_7.png)

![12_8](https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/12_8.png)

- adding ligand changes the thickness of biolayer, causing a spectrum shift. (interference, thus intensity changes)
- record the shift in different wavelengths. 
- extract an index from it as "response". (like the peak shift)
- collect this "response" as a function of time
- operation: just dip the tip into ligand/buffer solutions

#### sensorgrams

<img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/12_6.png" alt="12_6" style="zoom:20%;" />

$k_{on}$ and $k_{off}$ are fitted with $y=y_\infty+A\exp(-kt)$



## LSM3243 outline

1. protein structure

   - CORN
   - peptide planar
   - torsion angle definition
   - $\omega$ : trans, Pro
   - $\phi$ and $\psi$ : $\ce{N-C_{\alpha}}, \ce{C_O-C_{\alpha}}$
   - ramachandran plot
     - vdw radii, 3 regions
     - special: Gly, Pro. interrupt $\alpha, \beta$
   - structure features of typical conformations
     - helix, sheet, turn, coil
   - side chain
     - definition, $\chi_1, \chi_2$, other, last $\chi$.

2. interactions

   - electrostatic
     - effect of solvent/environment, salt, 
     - which parts carry charge
     - dipole: no charge. approximation. solvent with solute/solvent
   - 

3. nucleic acid

   - 

4. structure of membrane

   - lipids
     - features
       - amphiphilic (phospholipid, cholesterol)
       - differ in composition
       - asymmetric (two)
       - random (same)
     - types
       - glycerophospholipids: most, 13 C
       - sphingolipids: long and straight chain, stronger vdW to hold
       - cholesterol: hold more (rigid, two faces, hydrophobic)
     - lipid rafts
       - thicker
       - resistant
       - functional proteins
   - protein
     - intergral
     - peripheral
     - anchored
   - micelles
     - CMC, single layer--micelles--disrupted
     - DPC, triglyceride
   - 

5. membrane potential

6. membrane transport

7. CD

   - principle

   - applications

     - DNA conformation

     - protein secondary structure

     - protein folding under various conditions

       (folding: does it lose normal structure)

     - protein-ligand

     - protein mutation

     - folding kinetics

8. fluorescence

   - principle
     - basic
     - polarization
   - applications
     - interactions and binding parameters
     - protein folding under various conditions
     - visualization
   - FRET
     - conditions
     - study interactions (conformation)
   - quenching: study accessibility into aromatic

9. NMR

   - principle
     - basic, chemical shift, J coupling, T2
   - applications
     - inhibitor/drug identification
     - protein folding
     - metabonomics (biomarker)

   summary

10. comformation transition

    - model
      - basic and simple models
      - zipper model

11. protein folding

12. protein interaction



## final exam

1. a protein with 10 beta (barrel), 2 helix, no Cys/Trp

   1. possible interactions in helices? interaction that mainly hold the barrel? possible interactions between helix and helix-cap?
   2. reasons why 5% SDS makes unfolded
   3. design a protocol to verify it's unfolded
   4. how to determine if it interact with a 3<sub>10<sub/>-AA protein? if so, how to determine binding ratio?

2. a G<sub>12<sub/> and C<sub>12<sub/> dsDNA

   1. 60℃, no salt, what form? why?
   2. 1M NaCl, what form, why?
   3. what technique to use?

3. a POPS vesicle, liposome-like

   1. glucose, alanine, methanol. who can retain in the vesicle?

      i might forgot "polar"

   2. add 1 mM NaCl and 0.2 mM protein with 10 charge (Cl^-^ as counter ion) outside, calculate equilibrium concentration of all

4. two plots of v and [L]. method 1: measure the concentration of protein that bound on the ligand. method 2: measure the concentration of ligand that bound on the protein. data also given

   <img src="https://gitee.com/gxf1212/notes/raw/master/course/Physics-biophysics/molecular-biophysics-NUS/molecular-biophysics.assets/exam4.jpg" alt="exam4" style="zoom:33%;" />

   1. how many binding sites on the protein? why?

   2. $K_d$ value?

   3. $[L]$ at 5 mM, the fraction of the protein bounded?

   4. 0.9 of the protein bounded, $[L]$?

   5. draw and label the scatchard plot of method 2. 

      how to determine $n$ and $K_a$? is the sites equivalent?

   6. suggest technique for each method and explain.



## beyond course

#### Differential Scanning Fluorimetry (DSF)

- [差示扫描量热法 - 维基百科，自由的百科全书 (wikipedia.org)](https://zh.wikipedia.org/wiki/差示扫描量热法)
- [Differential Scanning Fluorimetry (DSF) | Center for Macromolecular Interactions (harvard.edu)](https://cmi.hms.harvard.edu/differential-scanning-fluorimetry)

#### more on NMR

radiofrequency pulse

free induction decay

when the nucleus magnet re-aligns, its direction changes and induces a chaning electric field. 

#### HSQC spectra

- [Wikipedia - Heteronuclear single quantum coherence spectroscopy](https://en.wikipedia.org/wiki/Heteronuclear_single_quantum_coherence_spectroscopy#HSQC_in_protein_NMR)
- https://www.protein-nmr.org.uk/solution-nmr/spectrum-descriptions/1h-15n-hsqc/

directly bonded atoms

#### protein-ligand

https://www.originlab.com/index.aspx?go=Solutions/CaseStudies&pid=1701











## Note

I'll use quotation box to highlight important content,<font color="grey"> grey text to give additional notes, </font><font color="skyblue"> other colored text to denote other environments,</font> <font color="yellowgreen"> so that it's clear.</font> 

Unimportant or very short part will be entitled "in addition" or inside a code box and a simple list. Do not spend much time on them.

<p><font color="grey" size=2.5> so that it's clear.</font></p>

<font color="yellowgreen"> The note is to help with reviewing rather than for publishing. </font>

<font color=\#8A2BE2> Do think and remember more than typing!!! </font>

The colors: https://blog.csdn.net/qq_40862304/article/details/109893291



```
平常有时间做一点，就为考试周减轻压力。有时间还可多看书。
```



> blubook主题可能为了图表的断页而压缩列表空间和字体。。
